packages/qwik/src/core/tests/inline-component.spec.tsx

```diff
@@ -419,4 +421,27 @@ describe.each([
       </Component>
     );
   });
+
+  it('should render component$ inside inlined wrapper', async () => {
+    const ComplexWrapper = (props: PublicProps<{}>, key: string | null, flags: number) => {
+      const cmpFn = component$(() => {
+        return <Slot />;
```

Comment:
If you render something else than the `<Slot />` here, I'm not sure it actually works as expected.

Here's an example:
```tsx
  it('should render component$ inside inlined wrapper', async () => {
    interface ComplexWrapperProps {
      foo: string;
    }

    const ComplexWrapper = (
      props: PublicProps<ComplexWrapperProps>,
      key: string | null,
      flags: number
    ) => {
      const cmpFn = component$<ComplexWrapperProps>(({ foo }) => {
        return (
          <div>
            {foo}: <Slot />
          </div>
        );
      });
      return cmpFn(props, key, flags);
    };

    const { vNode } = await render(<ComplexWrapper foo="bar">Test</ComplexWrapper>, { debug });
    expect(vNode).toMatchVDOM(
      <InlineComponent>
        <Component>
          <div>bar: Test</div>
        </Component>
      </InlineComponent>
    );
  });
```

Reply:
hey I changed to that test case, but everything is still working. The solution is a bit different than yours, so maybe this is the reason